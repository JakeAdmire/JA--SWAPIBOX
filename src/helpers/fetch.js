export const fetchHappens = async (url) => {
  try {
    const results = await fetch(url);
    const data = await results.json();
    return data.results;
  } catch(error) {
    return error;
  }
}