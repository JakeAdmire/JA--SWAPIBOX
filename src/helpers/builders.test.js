import { enzyme, shallow } from 'enzyme';
import React from 'react'

import { 
  peopleGatherer, 
  peopleCreator,
  planetsGatherer,
  planetsCreator,
  vehiclesGatherer,
  vehiclesCreator 
} from './builders';

import { fetchHappens } from './fetch';

describe('builders', () => {

  describe('peopleGatherer', () => {

    it('should return the correct data', async () => {

      let mockPerson = {
        birth_year: "19BBY",
        eye_color: "blue",
        films: [
          "https://swapi.co/api/films/2/",
          "https://swapi.co/api/films/6/",
          "https://swapi.co/api/films/3/",
          "https://swapi.co/api/films/1/",
          "https://swapi.co/api/films/7/"
        ],
        gender: "male",
        hair_color: "blond",
        height: "172",
        homeworld: "https://swapi.co/api/planets/1/",
        name: "Luke Skywalker",
        skin_color: "fair",
        species: ["https://swapi.co/api/species/1/"],
        starships: [
          "https://swapi.co/api/starships/12/",
          "https://swapi.co/api/starships/22/"
        ],
        vehicles: [
          "https://swapi.co/api/vehicles/14/",
          "https://swapi.co/api/vehicles/30/"
        ] 
      }

      const results = await peopleGatherer(mockPerson);
      const expected = {
        "category": "people",
        "homeworld": "Tatooine",
        "language": "Galactic Basic",
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
      expect(results).toEqual(expected);
    })

  })

  describe('planetsCreator', () => {

    let mockInfo = { 
      name: 'Earth',
      terrain: 'bumpy',
      population: 10000,
      climate: 'hot',
      residents: 3 
    };

    let wrapper = shallow( planetsCreator(mockInfo) );

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    }) 
    
  })

  describe('planetsGatherer', () => {

    it('should return the correct data', async () => {

      let mockPlanet = {
        climate: "temperate",
        created: "2014-12-10T11:35:48.479000Z",
        diameter: "12500",
        edited: "2014-12-20T20:58:18.420000Z",
        films: [
          "https://swapi.co/api/films/6/", 
          "https://swapi.co/api/films/1/"
          ],
        gravity: "1 standard",
        name: "Alderaan",
        orbital_period: "364",
        population: 2000000000,
        residents: [
          "https://swapi.co/api/people/5/", 
          "https://swapi.co/api/people/68/", 
          "https://swapi.co/api/people/81/"
          ],
        rotation_period: 24,
        surface_water: 40,
        terrain: "grasslands, mountains",
        url: "https://swapi.co/api/planets/2/" 
      }

      const results = await planetsGatherer(mockPlanet);
      const expected = {
        "category": "planets", 
        "climate": "temperate", 
        "name": "Alderaan", 
        "population": 2000000000, 
        "residents": 3, 
        "terrain": "grasslands, mountains"
      }
      expect(results).toEqual(expected);
    })

  })

  describe('vehiclesCreator', () => {

    let mockInfo = {
      model: "Digger Crawler",
      name: "Sand Crawler",
      passengers: "30",
      vehicleClass: "wheeled"
    }

    let wrapper = shallow( vehiclesCreator(mockInfo) );

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    }) 
    
  })
  
})