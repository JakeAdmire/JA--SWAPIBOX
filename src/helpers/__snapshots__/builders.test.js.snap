// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders planetsCreator should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="planet card"
  >
    <h3>
      Name: 
      Earth
    </h3>
    <p>
      Terrain: 
      bumpy
    </p>
    <p>
      Population: 
      10000
    </p>
    <p>
      Climate: 
      hot
    </p>
    <p>
      Residents: 
      3
    </p>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Name: 
          Earth
        </h3>,
        <p>
          Terrain: 
          bumpy
        </p>,
        <p>
          Population: 
          10000
        </p>,
        <p>
          Climate: 
          hot
        </p>,
        <p>
          Residents: 
          3
        </p>,
      ],
      "className": "planet card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "Earth",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "Earth",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Terrain: ",
            "bumpy",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Terrain: ",
          "bumpy",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            10000,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          10000,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Climate: ",
            "hot",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Climate: ",
          "hot",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Residents: ",
            3,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Residents: ",
          3,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Name: 
            Earth
          </h3>,
          <p>
            Terrain: 
            bumpy
          </p>,
          <p>
            Population: 
            10000
          </p>,
          <p>
            Climate: 
            hot
          </p>,
          <p>
            Residents: 
            3
          </p>,
        ],
        "className": "planet card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "Earth",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "Earth",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              "bumpy",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            "bumpy",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              10000,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            10000,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              "hot",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            "hot",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents: ",
              3,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents: ",
            3,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`builders vehiclesCreator should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="planet card"
  >
    <h3>
      Name: 
      Sand Crawler
    </h3>
    <p>
      Model: 
      Digger Crawler
    </p>
    <p>
      Class: 
      wheeled
    </p>
    <p>
      Passengers: 
      30
    </p>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Name: 
          Sand Crawler
        </h3>,
        <p>
          Model: 
          Digger Crawler
        </p>,
        <p>
          Class: 
          wheeled
        </p>,
        <p>
          Passengers: 
          30
        </p>,
      ],
      "className": "planet card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "Sand Crawler",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "Sand Crawler",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model: ",
            "Digger Crawler",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Model: ",
          "Digger Crawler",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class: ",
            "wheeled",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Class: ",
          "wheeled",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers: ",
            "30",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passengers: ",
          "30",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Name: 
            Sand Crawler
          </h3>,
          <p>
            Model: 
            Digger Crawler
          </p>,
          <p>
            Class: 
            wheeled
          </p>,
          <p>
            Passengers: 
            30
          </p>,
        ],
        "className": "planet card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "Sand Crawler",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "Sand Crawler",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model: ",
              "Digger Crawler",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model: ",
            "Digger Crawler",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class: ",
              "wheeled",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class: ",
            "wheeled",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers: ",
              "30",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers: ",
            "30",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
